added timeout flag to fix issue #68

ilan-pinto opened the pull request.

qiujian16 commented on pkg/cmd/join/cmd.go:
```diff
@@ -47,5 +47,6 @@ func NewCmd(clusteradmFlags *genericclioptionsclusteradm.ClusteradmFlags, stream
 	cmd.Flags().StringVar(&o.hubAPIServer, "hub-apiserver", "", "The api server url to the hub")
 	cmd.Flags().StringVar(&o.clusterName, "cluster-name", "", "The name of the joining cluster")
 	cmd.Flags().StringVar(&o.outputFile, "output-file", "", "The generated resources will be copied in the specified file")
+	cmd.Flags().StringVar(&o.timeout, "timeout", "", "extend timeout from 300 secounds ")
```
I think you could put it as a global flags here
https://github.com/open-cluster-management-io/clusteradm/blob/main/pkg/genericclioptions/clusteradm_flags.go#L23

and use IntVar so you would not need to convert from string to int.

qiujian16 commented on pkg/cmd/join/options.go:
```diff
@@ -20,6 +20,9 @@ type Options struct {
 	outputFile string
+
+	// override timeout for join 
+	timeout string
```
I think this should be removed?

ilan-pinto replied:
Fixed + added tests

qiujian16 commented:
pleass sign-off the commit

qiujian16 commented:
Hi @ilan-pinto you need to signoff the commit. Since you have 5 commits in this PR, you need to signoff each of them. OR you can squash then commit

qiujian16 commented:
/approve

ilan-pinto commented:
@qiujian16 am I missing something, why PR is not merged

qiujian16 commented:
it seems you push some new code. did you sign off the new commit?

ilan-pinto commented:
yes
check commit log

fixed PR comments

added tested for timeout

removed timout from Options.go

Signed-off-by: Ilan Pinto ***@***.***>


On Tue, Nov 30, 2021 at 8:51 AM Jian Qiu ***@***.***> wrote:

> it seems you push some new code. did you sign off the new commit?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/open-cluster-management-io/clusteradm/pull/77#issuecomment-982337063>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANWIKNPIHCX7VRHCOR4PPQ3UORYABANCNFSM5IZKNOEA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 

Ilan Pinto

Manager, Ecosystem Application Engineering

Red Hat <https://www.redhat.com/>
<https://www.redhat.com/>

qiujian16 commented:
/lgtm
